queue_services/business-pay/src/business_pay/services/gcp_auth.py

```diff
@@ -0,0 +1,55 @@
+# Copyright © 2024 Province of British Columbia
+#
+# Licensed under the BSD 3 Clause License, (the "License");
+# you may not use this file except in compliance with the License.
+# The template for the license can be found here
+#    https://opensource.org/license/bsd-3-clause/
+#
+# Redistribution and use in source and binary forms,
+# with or without modification, are permitted provided that the
+# following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice,
+#    this list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# 3. Neither the name of the copyright holder nor the names of its contributors
+#    may be used to endorse or promote products derived from this software
+#    without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS “AS IS”
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""GCP Auth Services."""
+from flask import current_app
+from google.oauth2 import id_token
+from google.auth.transport import requests
+from structured_logging import StructuredLogging
+
+
+logger = StructuredLogging.get_logger()
+
+
+def verify_gcp_jwt(flask_request):
+    """Verify the bearer token as sign by gcp oauth."""
+    try:
+        bearer_token = flask_request.headers.get("Authorization")
+        token = bearer_token.split(" ")[1]
+        audience = current_app.config.get("GOOGLE_CLIENT_ID")
+        claim = id_token.verify_oauth2_token(
+            token, requests.Request(), audience=audience
+        )
+        return claim
```

Comment:
Probably want to verify emails, that's what they did on their spec at least. 

https://cloud.google.com/pubsub/docs/authenticate-push-subscriptions

![image](https://github.com/bcgov/lear/assets/3484109/add727f3-36d6-4d3c-add2-3ae17eb35683)

Reply:
It's anote to check, they don't actually check and neither does namex from what I saw.

Reply:
Names checks.. I looked at the PR today and they do

Reply:
https://github.com/bcgov/namex/blob/22d209d77bb982f20d407e685bb9eb2918478694/services/pubsub/gcp_queue/gcp_auth.py#L28

Reply:
<img width="1180" alt="Screenshot 2024-05-13 at 8 27 27 AM" src="https://github.com/bcgov/lear/assets/22973013/f2a2ebf1-2440-4923-b72e-1c89bcfb3dbf">
did add the email check a week or two ago

Reply:
Yep, @seeker25 already put the link in.
I'll check.

Reply:
We have too much code copy/pasted and upgraded.
I'll add this to the gcp_queue package ... as soon as I get access back :(